common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java

```diff
@@ -213,8 +213,10 @@ private boolean processWebsiteRequest(@NonNull final WebView view, @NonNull fina
         final String methodName = "#processWebsiteRequest";
         final PackageHelper packageHelper = new PackageHelper(getActivity().getApplicationContext());
         if (url.startsWith(AuthenticationConstants.Broker.BROWSER_DEVICE_CA_URL)
-                && packageHelper.isPackageInstalledAndEnabled(AuthenticationConstants.Broker.COMPANY_PORTAL_APP_PACKAGE_NAME)) {
-            Logger.verbose(TAG + methodName, "It is a device CA request. Company Portal is installed.");
+                && packageHelper.isPackageInstalledAndEnabled(AuthenticationConstants.Broker.IPPHONE_APP_PACKAGE_NAME)
```

Comment:
This would break the existing scenario to check if Company Portal is installed or not right ? i.e. in mobile devices?

Reply:
+1

Reply:
Well, this was added to support their phone in general, and was an optimization for this link for the CP if their phone was not installed.

We don't have a better way of detecting that we're on their phone than by their app is installed on the device.

As long as there is a browser, this MDM CA link works by going to Intune Web Portal, and there is a link to the Google Play store there. So we'll 'regress' back to an extra click for Company Portal flows, we'll fix Microsoft Intune COBO flows from going to the wrong app, and we'll continue to work on Sakurai phones.